lxd/container_lxc.go

```diff
@@ -627,6 +628,10 @@ type containerLXC struct {
 	expiryDate time.Time
 }
 
+func (c *containerLXC) Type() string {
+	return instance.TypeContainer
```

Comment:
Shouldn't that also distinguish between container and container snapshot?

Reply:
@stgraber I'm not sure TBH, I didn't realise there were already defined types for containers and snapshots:

```
// Numerical codes for container types.
const (
	CTypeRegular  ContainerType = 0
	CTypeSnapshot ContainerType = 1
)
```

However, thinking about it, I think I would be inclined to stick with instance type being either container or VM, and then adding an `IsSnapshot()` function to the identifier instance (there is already such a function on the container implementation), that way if the logic you are dealing with doesn't differentiate between a snapshot or not then you wont need to care, and if you do then you can use the `IsSnapshot()` function rather than relying on internal DB codes.

Also, right now, AFAIK, no devices should be started when using snapshots.